packages/client-sdk-nodejs/src/preview-vector-index-client.ts

```diff
@@ -0,0 +1,52 @@
+import {ControlClient} from './internal/control-client';
+import {
+  AbstractVectorIndexClient,
+  IVectorIndexClient,
+  IVectorIndexControlClient,
+} from '@gomomento/sdk-core/dist/src/internal/clients';
+import {VectorIndexClientProps} from './vector-index-client-props';
+import {VectorDataClient} from './internal/vector-data-client';
+import {IVectorIndexDataClient} from '@gomomento/sdk-core/dist/src/internal/clients/vector/IVectorIndexDataClient';
+import {CacheConfiguration} from './config/configuration';
+import {defaultRetryStrategy} from './config/configurations';
+
+/**
+ * PREVIEW Vector Index Client
+ * WARNING: the API for this client is not yet stable and may change without notice.
+ *
+ * Vector and control methods return a response object unique to each request.
+ * The response object is resolved to a type-safe object of one of several
+ * sub-types. See the documentation for each response type for details.
+ */
+export class PreviewVectorIndexClient
+  extends AbstractVectorIndexClient
+  implements IVectorIndexClient
+{
+  constructor(props: VectorIndexClientProps) {
+    const controlClient: IVectorIndexControlClient = createControlClient(props);
+    const dataClient = createDataClient(props);
+    super(controlClient, dataClient);
+  }
+}
+
+function createControlClient(
+  props: VectorIndexClientProps
+): IVectorIndexControlClient {
+  return new ControlClient({
+    configuration: new CacheConfiguration({
+      loggerFactory: props.configuration.getLoggerFactory(),
+      transportStrategy: props.configuration.getTransportStrategy(),
+      retryStrategy: defaultRetryStrategy(
+        props.configuration.getLoggerFactory()
+      ),
+      middlewares: [],
+    }),
+    credentialProvider: props.credentialProvider,
+  });
```

Comment:
This seems like is for `cache` and we should change it to `vector`

Reply:
The `ControlClient` is internal, and currently houses the control plane operations for both vector and cache stuff. That is weird and it should probably be split up, but I didn't/don't think it's worth blocking on since it doesn't impact public API. wdyt?

Reply:
Got it. In Python we did split `_VectorIndexControlClient` which just uses the `vector_endpoint` from the `credentialProvider`; so seems like we could do the same here but I agree this is internal and can be split up at a later point. The configuration, however, I do think can be just changed to use the `VectorConfiguration` we created in this PR.

Reply:
i agree they should be split up when we have time.

The existing, internal ControlClient has a constructor signature that requires a CacheConfiguration as its first argument.

Reply:
I see. In that case i'm ok with this paritcularly because its not exposed to users.